### DISCUSSION TITLE
Can't install latest Firefox or Thunderbird without mentioning version in the installation command

### DISCUSSION THREAD

**Discussion by Witchilich** [microsoft/winget-pkgs#3327] [General]
Create issues related to the winget.exe client [here](https://github.com/microsoft/winget-cli/issues/new?template=Bug_Report.md)

# Brief description of your issue
Firefox and Thunderbird have there version names in app IDs. `winget install firefox` or `winget install thunderbird` fails, while `winget install Mozilla.firefox` or `winget install Mozilla.thunderbird` will install an older version.
![image](https://user-images.githubusercontent.com/56309157/91664668-5f599f80-eb0e-11ea-925a-9f689f726186.png)

# Steps to reproduce
Install Firefox or Thunderbird using normal installation command

# Expected behavior
Latest Firefox or Thunderbird will install using package installation.

# Actual behavior
winget will ask you to refine the input. Giving full app ID will only install an older version.
https://github.com/microsoft/winget-pkgs/tree/master/manifests/Mozilla/Firefox
https://github.com/microsoft/winget-pkgs/tree/master/manifests/Mozilla/Thunderbird

# Environment

```none
[winget --info]
Windows Package Manager v0.1.42241 Preview
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.18363.1016
Package: Microsoft.DesktopAppInstaller v1.10.42241.0

Any other software?
```

**Comment by OfficialEsco** [comment]
Ahhh i actually expected this to happen, this happens because someone totally fucked up the structure and they approved the PRs.

This is the wrong use of Manifests. The Manifest have a function for Localization and Language based Installers (which is not supported right now)
![image](https://user-images.githubusercontent.com/15158490/91706420-a8e1d300-eb7e-11ea-97f3-88ab4b8c107c.png)
![image](https://user-images.githubusercontent.com/15158490/91706454-b39c6800-eb7e-11ea-8129-1f9ba1d0f9a4.png)

**Comment by Witchilich** [comment]
Hey even changed the app id to `Mozilla.Firefox.Firefox_v80.0`.

**Comment by OfficialEsco** [comment]
Yup its totally fucked up and i have no idea how they could approve something like this...

There is probably noway of fixing this issue before all the Firefox v79.0 manifests are deleted, i'm working on adding the installers into the 79.0.yaml right now (even tho they are useless right now), this will take a while tho since there is 95 languages...

The Firefox v80.0 folder just contains the EN-US Version so that should be a easy fix tho, but thats just 1 out of 2 issues.

**Comment by Witchilich** [comment]
Well I made these PRs for the latest stable Firefox/Thunderbird
https://github.com/microsoft/winget-pkgs/pull/3339
https://github.com/microsoft/winget-pkgs/pull/3340

**Comment by denelon** [comment]
Unfortunately, multi-language/installer support isn't ready. Rather than "block" manifests for different languages, the decision was made to approve with languages. We knew we would have to clean up the extra manifests once proper support is in place. If you have suggestions on how we might address this in a more consistent manner, feel free to open up a discussion. We're interested in better approaches.

**Comment by OfficialEsco** [comment]
Firefox/Thunderbird contains 95 languages, i doubt anyone REQUIRES their native language in WinGet Alpha 0.1. None will go trough 95 languages for every update, so why break the download for just one Firefox/Thunderbird version?

How to address; Do like normal, use one mainfest, add the installers for Multi-Language support for whenever it gets implemented.
If everyone adds every language, 32bit and 64bit manifest's WinGet would be painfull to use.

If a language is very popular or needed you can do like Firefox-cs.

**Comment by Witchilich** [comment]
For starters the en-US version should be in "Mozilla/Firefox" or "Mozilla/Thunderbird" directory and the app ID should just be `Mozilla.Firefox` or `Mozilla.Thunderbird`. Keep the other languages in the version directories till you figure out multi-language manifest implementation.

**Comment by Suirtimed** [comment]
We're close to sharing a draft and discussion for multi-language support. It's being written up, and samples are being created so we can share for review.

**Reply by OfficialEsco** [reply]
It seems like you guys need to fix the error happening on #3342 #3344 #3345 #3346 first tho, it looks like a Manifest with 95 languages are too long for the System to handle. (Unless it dosn't like the 1050 commented out lines)
